Add Deck Settings

Josh5 opened the pull request:
<!--
    READ ALL COMMENTS IN THIS TEMPLATE BEFORE SUBMITTING YOUR PR!

    This template is for adding a new plugin to the store. If you are taking any other action, please start over by creating a new pull request and selecting the appropriate template.

    If you have not already, please read the review and testing page on the wiki.
    https://wiki.deckbrew.xyz/plugin-dev/review-and-testing

    Before submitting, make sure you have done the following:
    - Tested two other plugins and left feedback on their PRs as described below.
    - Replaced REPLACE_WITH_PLUGIN_NAME and REPLACE_WITH_SUMMARY with the appropriate information.
    - Filled out the Task Checklist, including the Yes/No questions.
    - Deleted the unnecessary testing checkbox.
-->

# Deck Settings

<!--
    Include a detailed summary of what the plugin does, attaching images or videos if necessary. If your plugin has similar functionality to another plugin on the store, explain how it differs and why your plugin should be allowed on the store.
-->

This plugin integrates with the Deck Settings API (https://deckverified.games) to fetch game compatibility and configuration reports. These reports are sourced from the community-driven repository at https://github.com/DeckSettings/game-reports-steamos. The plugin allows users to easily view game-specific performance tweaks, compatibility information, and configuration suggestions directly on their Steam Deck, streamlining the process of optimizing game settings for handheld devices.

Features:
- Fetches and displays game reports tailored for handheld gaming devices like the Steam Deck, ROG Ally, and others.
- Provides quick access to community-verified configuration tips, graphics settings, and performance tweaks.
- Leverages the Deck Settings API to dynamically retrieve the latest game data.
- Allows users to search for game configurations by name or App ID.

This is the first submission of the plugin to the Decky plugin repository.

## Task Checklist

<!--
    For checkboxes, change [ ] to [x] to check the box.
    For Yes/No questions, replace "Yes/No" with "Yes" or "No".
-->

### Developer

- [X] I am the original author or an authorized maintainer of this plugin.
- [X] I have abided by the licenses of the libraries I am utilizing, including attaching license notices where appropriate.

### Plugin

- [X] I have verified that my plugin works properly on the Stable and Beta update channels of SteamOS.
- [X] I have verified my plugin is unique or provides more/alternative functionality to a plugin already on the store.

### Backend

- **No**: I am using a custom backend other than Python.
- **No**: I am using a tool or software from a 3rd party FOSS project that does not have it's dependencies [statically linked](https://en.wikipedia.org/wiki/Static_library).
- **No**: I am using a custom binary that has all of it's dependencies statically linked.

### Community

<!--
    Please submit both of your testing reports before creating your PR. You will need to link the comments you left in a comment on this PR.

    If no plugin additions or updates are ready for testing at this time, then you may ignore this checkbox. You may be asked to test new plugins as they are submitted.
-->

- [ ] I have tested and left feedback on two other [pull requests][pulls] for new or updating plugins.
- [ ] I have commented links to my testing report in this PR.

## Testing

<!--
    DO NOT FORGET THIS STEP! Otherwise, testers may incorrectly test your plugin.

    If your plugin uses the provided Python backend and React frontend, your plugin must be tested on the Stable or Beta update channel of SteamOS. REMOVE the line with the Preview checkbox below.

    If your plugin uses a custom backend or pre-build binaries without statically linked dependency (ex. glibc), your plugin must be tested on the SteamOS Preview update channel. REMOVE the line with the Stable or Beta checkbox below.
-->

- [x] Tested on SteamOS Stable or Beta update channel.

[pulls]: https://github.com/steamdeckHomebrew/decky-plugin-database/pulls?q=is%3Apr+is%3Aopen+sort%3Acreated-desc+-status%3Afailure+-draft%3Atrue+-author%3A%40me

TrainDoctor reviewed (changes requested):
Please update the license file with information or if you decide to use another license place the text for that license above the original in license file. Please delete the py_modules folder if you are not making use of it. The same for main.py if you're not utilizing the python backend as well.
Please update the description in package.json.
Please remove the flags from plugin.json and add plugin tags that match your plugin. Since you're plugin is not the template or using root, you can remove both of those tags. I don't see a reference to gamesearch component but the file is empty, is this for future use?
Please make these changes and your plugin can go to testing, thanks.

Josh5 commented:
The GameSearch component completely sliped in there. Was replaced when I did the SearchResultsView.
Thanks for these details. Seems I missed a lot of things.

- Deleted empty GameSearch componenet file.
- Updated the `LICENSE` file with myself as an author. Keeping license as BSD 3-Clause License.
- Removed the `py_modules` directory.
- Removed the `main.py`.
- Updated the description in `package.json` (and `README.md`).
- Removed flags and updated tags in `plugin.json`.

TrainDoctor commented:
@Josh5 please rebase on main to get rid of the playcount merge

Josh5 commented:
@TrainDoctor I have rebased to main. Thanks.

xXJSONDeruloXx commented:
everything seems to work well, though most games I tried said 0 reports. I suppose it's more of an issue with the site itself. Great plugin!

Josh5 commented:
Some good points raised. 

At the moment, the site and database are only a few weeks old. I imagine as it gets used more, the number of community supplied reports will grow.

I will look into developing a method of easily submitting reports from the plugin in the future. At the moment, I'd recommend using a companion mobile view via the site or even the official github app for submitting reports.

EMERALD0874 commented:
@Josh5 Your PR has been updated to use the latest version of the PR template. Please see the Community section of the Task Checklist regarding testing two other PRs to get the quickest response from our team. There are additional comments in the Markdown when you edit the PR text which you may find helpful. Thanks!
